src/Propel/Generator/Builder/Om/ObjectBuilder.php

```diff
@@ -3000,12 +3000,14 @@ protected function addToArray(string &$script): void
         }
         $script .= "
      *
-     * @return array|string An associative array containing the field names (as keys) and field values
+     * @throw \Propel\Runtime\Exception\PropelException
+     *
+     * @return array An associative array containing the field names (as keys) and field values
      */
     public function toArray(string \$keyType = TableMap::$defaultKeyType, bool \$includeLazyLoadColumns = true, array \$alreadyDumpedObjects = []" . ($hasFks ? ', bool $includeForeignObjects = false' : '') . ")
     {
         if (isset(\$alreadyDumpedObjects['$objectClassName'][\$this->hashCode()])) {
-            return '*RECURSION*';
+            throw new PropelException('*RECURSION*');
```

Comment:
Couldnt we just return `['*RECURSION*']`?
I dont think throwing exception works here.

Reply:
Ok. But for me, it looks like an exceptional case.